### DISCUSSION TITLE
password authentication failed for user <username>. Role <username> does not exist

### DISCUSSION THREAD

**Discussion by dyaipayan22** [docker-library/postgres#1253] [General]
I am trying to run postgres using docker-compose. On migrating with drizzle, I can preview the tables in drizzle-studio but getting this error when trying to check a POST request using postman. According to the discussions earlier, I tried deleting volumes but it did not help. I do not have postgres installed on my windows which could have been a reason for the error.

docker-compose.yml
![image](https://github.com/docker-library/postgres/assets/115914025/9a33386b-2dcf-4147-bac5-ab4185553ee8)

logs
![image](https://github.com/docker-library/postgres/assets/115914025/9360ea8a-6282-4215-8433-4644ed5210e8)

**Comment by LaurentGoderre** [comment]
It looks to me like something is trying to connect to the db in the container using a user that doesn't exist

**Reply by dyaipayan22** [reply]
@LaurentGoderre I don;t have any other instance running. Also, I have used this same docker-compose in a nextjs project, it works perfectly there. I was able to do all sort of requests through the application (haven't tested out endpoints on postman). What can be the possible issue and where should I debug?

**Reply by LaurentGoderre** [reply]
You can run `docker-compose exec -it -u postgres product-db psql -c '\du'` to see the liost of users in the db and if `dyaipayan` is there

**Comment by imrim12** [comment]
On Windows, if you installed PGAdmin locally from an installer, you will probably have PostgreSQL installed as well, make sure you uninstall it and run everything from Docker only

![image](https://github.com/user-attachments/assets/61cdea20-531d-4aaf-9ce9-0ec54aa3438e)
